contracts/multisig/src/signing.rs

```diff
@@ -73,33 +73,51 @@ pub fn validate_session_signature(
         });
     }
 
-    let valid = sig_verifier.map_or_else(
-        || signature.verify(&session.msg, pub_key),
-        |verifier| {
-            verifier
-                .verify_signature(
+    sig_verifier
+        .map_or_else(
+            || signature.verify(&session.msg, pub_key),
+            |sig_verifier| {
+                call_sig_verifier(
```

Comment:
is there a specific reason why you extracted this? I think in this case the extra overhead is not worth it with so many parameters, just to convert the bool to an error

Reply:
I thought the `map_or_else` would be easier to read without too much clutter in there

Reply:
ok good point!